### DISCUSSION TITLE
Is Compensable activties supported in V3?

### DISCUSSION THREAD

**Discussion by defunky** [elsa-workflows/elsa-core#5652] [Q&A]
I can't find any information on Compensable activities in v3, I'm assuming there's a different approach that your suppose to do in V3?

**Comment by glime-ai** [comment]
In Elsa 3, the approach to handling compensations and related activities has evolved from the previous version. While specific "Compensable" activities from Elsa 2 may not be directly available in Elsa 3, you can achieve similar functionality using the new features and activities provided by Elsa 3.

Elsa 3 introduces a new set of activities and concepts that can be used to handle scenarios where compensations might be necessary. For example, you can use the combination of "Try/Catch/Finally" activities to implement try-catch-finally blocks within your workflows, which can serve as a way to handle exceptions and perform compensatory actions.

If you are looking for a direct replacement for the "Compensable" activity from Elsa 2, it's important to review the Elsa 3 documentation and explore the new activities and patterns that are available. Since Elsa 3 is a complete rewrite with many improvements and changes, some activities from Elsa 2 may have been replaced with more flexible or different mechanisms.

For detailed guidance on how to implement compensation logic in Elsa 3, you should consult the official documentation or reach out to the Elsa community for advice. Additionally, you can look into creating custom activities if the built-in ones do not meet your specific requirements.

If you have a specific use case or scenario in mind, please provide more details so that I can offer more targeted advice or guidance.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by electrocnic** [comment]
@sfmskywalker

Do you have any updates or information on this? I searched through the repo and documentation and Elsa Studio and could not find any recommended way or feature that covers compensation in any way in Elsa 3, I could only find compensation for Elsa 2.
I also found another thread where you mentioned in #4280 that you are still working on a compensation mechanic in Elsa 3.

I am new to Elsa and so far this is the only library that stands out for what I need, in comparison to many others, so thank you so much for doing this!

**Reply by sfmskywalker** [reply]
Hi @electrocnic 

No update yet - work has yet to begin on compensable activities. There are no concrete timelines as of now. Our current focus lies with stability, reliability and prioritized features from customers. That said, it does help to know there are other users who are in need of this capability, which is something I'll communicate back internally.

Thank you for the kind words, I am happy to hear Elsa meets your needs!

**Reply by electrocnic** [reply]
thanks for the quick and informative response!
Then I can find workarounds in the meantime (implementing my own compensation logic somehow).

**Reply by sfmskywalker** [reply]
Anytime! Yes, you can totally implement your own version thanks to the compostable activity architecture. And, bonus points for submitting a PR in case you manage to implement it correctly 😃
